### DISCUSSION TITLE
Simplify the way createSelectors works

### DISCUSSION THREAD

**Discussion by vcardins** [pmndrs/zustand#2263] [Ideas]
I wonder if it's ok to simplify the [createSelectors](https://github.com/pmndrs/zustand/blob/a4578fa5c48c013408744eddd7d051ade223d076/docs/guides/auto-generating-selectors.md#create-the-following-function-createselectors) function to return the actual state value instead of another function.

I tweaked as follow (notice that I removed the `() =>` from `... { use: { [K in keyof T]: () => T[K] }` ):

```
import { StoreApi, useStore } from 'zustand';

type WithSelectors<S> = S extends { getState: () => infer T }
	? S & { use: { [K in keyof T]: T[K] } }
	: never

export const createSelectors = <S extends StoreApi<object>>(_store: S) => {
	const store = _store as WithSelectors<typeof _store>;
	const stateKeys = Object.keys(store.getState());
	const use = stateKeys.reduce((result, k) => ({
		...result,
		[k]: useStore(_store, (s) => s[k as keyof typeof s]),
	}), {} as Record<string, any>);

	return { ...store, use };
};
```
Thanks

**Comment by dai-shi** [comment]
I don't think you can use the hook outside React.

**Reply by vcardins** [reply]
I didn't mean to use it outside react. Based on the example of the documentation, we'd have `store.use.items()`, however items it's an array. What I'm proposing is to access it by its original type: `store.use.items`.

**Reply by dbritto-dev** [reply]
@vcardins that's not possible since what you are calling under the hood is a custom hook

**Reply by vcardins** [reply]
@dbritto-dev Thanks for the answer. I'll play with it a little more and provide my findings.
